pkg/client/cache/cache.go

```diff
@@ -10,43 +10,43 @@ import (
 	"github.com/telepresenceio/telepresence/v2/pkg/filelocation"
 )
 
-// ensureCacheDir returns the full path to the directory "telepresence", parented by the directory
-// returned by UserCacheDir(). The directory is created if it does not exist.
-func ensureCacheDir(ctx context.Context) (string, error) {
-	cacheDir, err := filelocation.AppUserCacheDir(ctx)
-	if err != nil {
-		return "", err
-	}
-	if err := os.MkdirAll(cacheDir, 0o700); err != nil {
-		return "", err
-	}
-	return cacheDir, nil
-}
-
 func SaveToUserCache(ctx context.Context, object any, file string) error {
 	jsonContent, err := json.Marshal(object)
 	if err != nil {
 		return err
 	}
-	dir, err := ensureCacheDir(ctx)
+
+	// get base path of user cache
+	cacheDir, err := filelocation.AppUserCacheDir(ctx)
 	if err != nil {
 		return err
 	}
-	return os.WriteFile(filepath.Join(dir, file), jsonContent, 0o600)
+	// add file path (ex. "ispec/00-00-0000.json")
+	fullFilePath := filepath.Join(cacheDir, file)
+	// get dir of joined path
+	dir := filepath.Dir(fullFilePath)
+	if err := os.MkdirAll(dir, 0o700); err != nil {
+		return err
+	}
+	return os.WriteFile(fullFilePath, jsonContent, 0o600)
 }
 
 func LoadFromUserCache(ctx context.Context, dest any, file string) error {
 	dir, err := filelocation.AppUserCacheDir(ctx)
 	if err != nil {
 		return err
 	}
-	path := filepath.Join(dir, file)
-	jsonContent, err := os.ReadFile(path)
+	filePath := filepath.Join(dir, file)
+	return LoadFromFilePath(dest, filePath)
+}
+
+func LoadFromFilePath(dest any, filePath string) error {
```

Comment:
This doesn't have to be public does it? if the idea is that SaveToUserCache will accept something like "ispect/foo-bar.json", then LoadFromUserCache should accept the same and there's no need to ever resort to an absolute path; I say we make this private

Reply:
It has to be public so I can use it from the pro-repo 🤷